node/config/load.go

```diff
@@ -46,7 +72,51 @@ func FromReader(reader io.Reader, def interface{}) (interface{}, error) {
 	return cfg, nil
 }
 
-func ConfigUpdate(cfgCur, cfgDef interface{}, comment bool) ([]byte, error) {
+// function from raw config string to loadability
+type loadabilityCheck struct {
+	configOk       func(string) error
+	configNotFound func() error
+}
+
+func DefaultFullNodeLoadabilityCheck() loadabilityCheck {
+	configOk := func(cfgRaw string) error {
+		poisonStr := "#EnableSplitstore =" // splitstore must be set explicitly
+		if strings.Contains(cfgRaw, poisonStr) {
+			return xerrors.Errorf("Config contains disallowed string %s, refusing to load. Please explicitly set EnableSplitstore. Set it to false to keep chain management unchanged from March 2023 set it to true if you want to discard old state by default", poisonStr)
+		}
+		return nil
+	}
+	configNotFound := func() error {
+		return xerrors.Errorf("FullNode config not found and fallback to default disallowed.  Set up config and explicitly set EnableSplitstore.  Set it to false to keep chain management unchanged from March 2023 set it to true if you want to discard old state by default ")
+	}
+	return loadabilityCheck{
+		configOk:       configOk,
+		configNotFound: configNotFound,
+	}
+}
+
+func LoadabilityCheckNone() loadabilityCheck {
+	return loadabilityCheck{
+		configOk:       func(string) error { return nil },
+		configNotFound: func() error { return nil },
+	}
+}
+
+type keepUncommented func(string) bool
+
+func DefaultFullNodeCommentFilter() keepUncommented {
+	return func(s string) bool {
+		return strings.Contains(s, "EnableSplitstore")
+	}
+}
+
+func CommentFilterNone() keepUncommented {
+	return func(string) bool {
+		return false
+	}
+}
+
+func ConfigUpdate(cfgCur, cfgDef interface{}, comment bool, filter keepUncommented) ([]byte, error) {
```

Comment:
Docs. Also, please use something like the functional options pattern.

Reply:
Added a small amount of docs and one functional option related to what I'm doing.